### ISSUE TITLE
FileBackedVisualRecord

### ISSUE THREAD

**Issue by ghost** [dchaplinsky/visual-logging#11]
Hi

visual-logging is fantastic, but saving images directly inside html source code doesn't scale well past 100 images or so - htmls just take ages to load.
Sometimes I find I would really like to log many images, and for such situations I wrote a VisualRecord version that saves images to file system and puts their paths in src attribute of img tag instead. Would you be open to a pull request for this feature?

Here's code I'm using now, just for reference:
```python
class FileBackedVisualRecord:
    """
    visual-logging inspired class that can be used to log images.
    Saves images in specified folder, giving them random names
    """

    def __init__(self, title, images, directory, footnote=None):
        """
        Constructor
        :param title: string
        :param images: list of numpy images
        :param directory: str, path to directory in which log will be saved.
        Images will be saved to directory /images inside that directory. /images directory should be created by client
        :param footnote: string or None
        """

        self.title = title
        self.images = images
        
        images_directory = os.path.join(directory, "images")

        if not os.path.isdir(images_directory):
            raise ValueError("No directory {} found".format(images_directory))

        self.directory = directory
        self.footnote = footnote if footnote else ""

    def __repr__(self):

        style = 'style="margin: 1em;"'

        images_htmls = []

        image_template = jinja2.Template('<img src="{{ path }}" {{ style }}>')

        for image in self.images:

            html_path = os.path.join("images", str(uuid.uuid4()) + ".jpg")

            file_path = os.path.join(self.directory, html_path)
            cv2.imwrite(file_path, image)

            html = image_template.render(path=html_path, style=style)

            images_htmls.append(html)

        template = jinja2.Template(
            "<details open>"
            "<summary><h1>{{ title }}</h1></summary>"
            "<div>{{ images_template }}</div><br>"
            "{{ footnote }}<hr>"
            "</details>"
        )

        return template.render(
            title=self.title,
            images_template="".join(images_htmls),
            footnote=self.footnote,
        )

```

**Comment by dchaplinsky** [comment]
Please do a PR but make sure that tests are passing

**Comment by ghost** [comment]
Great, I'll send a pull request then when I'll find time to adjust code to visual logging's style, thank you

**Comment by ghost** [comment]
@dchaplinsky 
I tried to start working on this issue, and as a first step tried to spin out a docker container with python 3.8, and install devrequirements.txt, but installation failed on `matplotlib==1.4.1`.
I can go forward bumping up matplotlib to a more recent version (e.g. `3.3.1`), but all of this begs a question on what versions do you want code to support.

Looking at `.travis.yml` definition it seems to be you test only on `python 2.7`, but it's no longer supported.
( https://www.python.org/doc/sunset-python-2/ ).

`python 3.5` is reaching its end of life this month `https://www.python.org/downloads/release/python-3510/`

So in short what python versions should a pull request target?

**Comment by dchaplinsky** [comment]
Just pick all supported versions of python, please. I suppose, it's 3.6+

**Comment by PuchatekwSzortach** [comment]
@dchaplinsky 
I couldn't get tests to run with python 3.8
`nosetests -vs vlogging/tests/corner_cases.py vlogging/tests/basic.py --with-isolation --with-coverage --cover-erase --cover-package=.`

Digging around a bit it seems to me this is because tests try to modify `sys.path`:
- vlogging/tests/corner_cases.py inserts `dummies` package into `sys.path` before importing `vlogging` to test renderers registration in `vlogging/__inti__.py` fails gracefully
- vlogging/tests/basic.py tries to pop `dummies` packages form path before importing `vlogging`, so it has real renderers available

but code inside `vlogging.__init__` is only ever imported by python once, even if `import vlogging` appears in both modules - verified this by adding some debug statements inside there.

Was is maybe the case that with Python 2.7 importing a module twice caused its `__init__.py` to be executed twice? In that case renderers would be recomputed correctly inside each test module, but it doesn't seem to work that way in Python 3.
